src/library/dao/autodjcratesdao.cpp

```diff
@@ -58,11 +58,11 @@ int bounded_rand(int highest) {
 
 AutoDJCratesDAO::AutoDJCratesDAO(
         int iAutoDjPlaylistId,
-        TrackCollection* pTrackCollection,
+        TrackCollectionManager* pTrackCollectionManager,
         UserSettingsPointer pConfig)
         : m_iAutoDjPlaylistId(iAutoDjPlaylistId),
-          m_pTrackCollection(pTrackCollection),
-          m_database(pTrackCollection->database()),
+          m_pTrackCollectionManager(pTrackCollectionManager),
+          m_database(pTrackCollectionManager->internalCollection()->database()),
```

Comment:
This accesses the internal collection from outside of the track collection manager. should we add a method to retrieve the db directly?

Reply:
No. TrackCollectionManager does not store a database connection. And we should not introduce such a dependency just for those legacy components.

Leaking the connection from TrackCollection is already wrong, together with direct access to the internal DAOs. Let's keep it as is until someone contributes a rewrite of the database access layer. I don't expect that this will ever happen.